packages/theme/src/cli/utilities/asset-file-syncer.ts

```diff
@@ -186,3 +199,22 @@ async function partitionFilesByReconciliationStrategy(files: {
 
   return {localFilesToDelete, filesToDownload, filesToUpload, remoteFilesToDelete}
 }
+
+function pollThemeEditorChanges(targetTheme: Theme, session: AdminSession, localThemeFileSystem: ThemeFileSystem) {
```

Comment:
I tried to write a debounce here but kept getting stuck with all the TS async signatures, etc

@karreiro, could I get some help on this + on testing?

Reply:
Could probably use this instead

https://github.com/Shopify/cli/blob/eb307f47b6a5d470174b8ddfc401d83d6e26ca08/packages/cli-kit/src/public/common/function.ts#L38-L44